src/components/CustomerGroups/index.js

```diff
@@ -16,14 +16,19 @@ class CustomerGroups extends Component {
   };
 
   componentDidMount() {
-    const { type } = this.state;
+    const { role } = this.props;
+    let { type } = this.state;
+    if (role !== undefined) {
+      type = role[0];
+    }
```

Comment:
what is it?
 this.setState

you can"t assign some value to state like this
` type = role[0]`